lib/Controller/RoomController.php

```diff
@@ -1316,7 +1301,25 @@ public function addParticipantToRoom(string $newParticipant, string $source = 'u
 			return new DataResponse([], Http::STATUS_BAD_REQUEST);
 		}
 
-		return new DataResponse();
+		// attempt adding the listed users to the room
+		// existing users with USER_SELF_JOINED will get converted to regular USER participants
+		foreach ($participantsToAdd as $index => $participantToAdd) {
+			$existingParticipant = $participantsByUserId[$participantToAdd['actorId']] ?? null;
+
+			if ($existingParticipant !== null) {
+				unset($participantsToAdd[$index]);
```

Comment:
potential frown here ? I didn't want to make yet another copy of the array... and apparently PHP allows for that, so yay!